tests/Doctrine/Tests/DBAL/Functional/Schema/SchemaManagerFunctionalTestCase.php

```diff
@@ -1452,4 +1456,37 @@ public function testCreateAndListSequences() : void
         self::assertEquals($sequence2AllocationSize, $actualSequence2->getAllocationSize());
         self::assertEquals($sequence2InitialValue, $actualSequence2->getInitialValue());
     }
+
+    /**
+     * @group #3086
+     */
+    public function testComparisonWithAutoDetectedSequenceDefinition() : void
+    {
+        if (! $this->_sm->getDatabasePlatform()->supportsSequences()) {
+            self::markTestSkipped('This test is only supported on platforms that support sequences.');
+        }
+
+        $sequenceName           = 'sequence_auto_detect_test';
+        $sequenceAllocationSize = 5;
+        $sequenceInitialValue   = 10;
+        $sequence               = new Sequence($sequenceName, $sequenceAllocationSize, $sequenceInitialValue);
+
+        $this->_sm->dropAndCreateSequence($sequence);
+
+        $createdSequence = array_values(
+            array_filter(
+                $this->_sm->listSequences(),
+                function (Sequence $sequence) use ($sequenceName) : bool {
+                    return strcasecmp($sequence->getName(), $sequenceName) === 0;
```

Comment:
Things in Oracle are uppercase... Changed to case insensitive comparison. @morozov any pointers how to do this better, without array_filter?

Reply:
No idea given the current API. We could improve the API to make it return sequences indexed by name.

Reply:
BTW, I found it literally a couple of days ago that DB2 supports [`DB2_CASE_LOWER`](http://php.net/manual/en/ibm-db2.constants.php#constant.db2-case-lower), but we use driver level case conversion only with PDO. If the same thing existed for Oracle, we could throw away case conversion from the portability layer.